packages/flutter/lib/src/cupertino/tab_scaffold.dart

```diff
@@ -289,43 +292,80 @@ class CupertinoTabScaffold extends StatefulWidget {
   /// Defaults to true and cannot be null.
   final bool resizeToAvoidBottomInset;
 
+  /// Restoration ID to save and restore the state of the [CupertinoTabScaffold].
+  ///
+  /// This property only has an affect when no [controller] has been provided:
+  /// If it is non-null (and no [controller] has been provided), the scaffold
+  /// will persist and restore the currently selected tab index. If a
+  /// [controller] has been provided, it is the responsibility of the owner of
+  /// that controller to persist and restore it, e.g. by using a
+  /// [RestorableCupertinoTabController].
```

Comment:
Are there a bunch of new restorable versions of classes now to support state restoration?  This is the first state restoration PR I've taken a close look at.

Reply:
There are a few. Another example is the RestorableTextEditingController.

Reply:
Ah thanks, will study up.